app/DAO/BakeryDatabase.scala

```diff
@@ -89,60 +90,76 @@ class BakeryDatabase @Inject() (
-        sql"""UPDATE product SET name = $newName, quantity = $newQty, price = $newPrice, updated_at = $newStamp WHERE id::text = $id"""
-    } else if (newName.isDefined && newQty.isDefined && newPrice.isEmpty) {
-      query =
-        sql"""UPDATE product SET name = $newName, quantity = $newQty, updated_at = $newStamp WHERE id::text = $id"""
-    } else if (newName.isDefined && newQty.isEmpty && newPrice.isDefined) {
-      query =
-        sql"""UPDATE product SET name = $newName, price = $newPrice, updated_at = $newStamp WHERE id::text = $id"""
-    } else if (newName.isEmpty && newQty.isDefined && newPrice.isDefined) {
-      query =
-        sql"""UPDATE product SET quantity = $newQty, price = $newPrice, updated_at = $newStamp WHERE id::text = $id"""
-    } else if (newName.isDefined && newQty.isEmpty && newPrice.isEmpty) {
-      query =
-        sql"""UPDATE product SET name = $newName, updated_at = $newStamp WHERE id::text = $id"""
-    } else if (newName.isEmpty && newQty.isDefined && newPrice.isEmpty) {
-      query =
-        sql"""UPDATE product SET quantity = $newQty, updated_at = $newStamp WHERE id::text = $id"""
-    } else {
-      query =
-        sql"""UPDATE product SET price = $newPrice, updated_at = $newStamp WHERE id::text = $id"""
+
+    (newName, newQty, newPrice) match {
+      case (Some(name), Some(quantity), Some(price)) =>
+        sql"""UPDATE product SET name = $name, quantity = $quantity, price = $price, updated_at = $newStamp WHERE id::text = $id""".update.run
+          .transact(bakeryTransactor.xa)
+          .unsafeRunSync()
+      case (Some(name), Some(quantity), None) =>
+        sql"""UPDATE product SET name = $name, quantity = $quantity, updated_at = $newStamp WHERE id::text = $id""".update.run
+          .transact(bakeryTransactor.xa)
+          .unsafeRunSync()
+      case (Some(name), None, Some(price)) =>
+        sql"""UPDATE product SET name = $name, price = $price, updated_at = $newStamp WHERE id::text = $id""".update.run
+          .transact(bakeryTransactor.xa)
+          .unsafeRunSync()
+      case (None, Some(quantity), Some(price)) =>
+        sql"""UPDATE product SET quantity = $quantity, price = $price, updated_at = $newStamp WHERE id::text = $id""".update.run
+          .transact(bakeryTransactor.xa)
+          .unsafeRunSync()
+      case (Some(name), None, None) =>
+        sql"""UPDATE product SET name = $name, updated_at = $newStamp WHERE id::text = $id""".update.run
+          .transact(bakeryTransactor.xa)
+          .unsafeRunSync()
+      case (None, Some(quantity), None) =>
+        sql"""UPDATE product SET quantity = $quantity, updated_at = $newStamp WHERE id::text = $id""".update.run
+          .transact(bakeryTransactor.xa)
+          .unsafeRunSync()
+      case (None, None, Some(price)) =>
+        sql"""UPDATE product SET price = $price, updated_at = $newStamp WHERE id::text = $id""".update.run
+          .transact(bakeryTransactor.xa)
+          .unsafeRunSync()
```

Comment:
notice you are running 
```scala
    .update
    .run
    .transact(bakeryTransactor.xa)
    .unsafeRunSync()
```
on every part of the forks. you can assign a variable to the query instead and then run things at the end:

```scala
val query = (newName, newQty, newPrice) match {
/// rest of the code
}

query
    .update
    .run
    .transact(bakeryTransactor.xa)
    .unsafeRunSync()
```

to make your life easier to modify in the future. You'll most likely see this pattern as well.